go-client/client/table_stub.go

```diff
@@ -0,0 +1,469 @@
+			return nil, err
+		}
+
+		return parseCreationResponse(ts.client, resp)
+	} else {
+		return nil, errors.New("no table by the name " + name + " (maybe it isn't synced?)")
+	}
+}
+
+// EmptyTableQuery is like EmptyTable, except it can be used as part of a query.
+func (ts *tableStub) EmptyTableQuery(numRows int64) QueryNode {
+	qb := newQueryBuilder(ts.client, nil)
+	qb.ops = append(qb.ops, emptyTableOp{numRows: numRows})
+	return qb.curRootNode()
+}
+
+// EmptyTable creates a new empty table in the global scope.
+//
+// The table will have zero columns and the specified number of rows.
+func (ts *tableStub) EmptyTable(ctx context.Context, numRows int64) (*TableHandle, error) {
+	ctx, err := ts.client.withToken(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	result := ts.client.newTicket()
+
+	req := tablepb2.EmptyTableRequest{ResultId: &result, Size: numRows}
+	resp, err := ts.stub.EmptyTable(ctx, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return parseCreationResponse(ts.client, resp)
+}
+
+// TimeTableQuery is like TimeTable, except it can be used as part of a query.
+func (ts *tableStub) TimeTableQuery(period int64, startTime *int64) QueryNode {
+	var realStartTime int64
+	if startTime == nil {
+		// TODO: Same question as for TimeTable
+		realStartTime = time.Now().UnixNano()
+	} else {
+		realStartTime = *startTime
+	}
+
+	qb := newQueryBuilder(ts.client, nil)
+	qb.ops = append(qb.ops, timeTableOp{period: period, startTime: realStartTime})
+	return qb.curRootNode()
+}
+
+// TimeTable creates a ticking time table in the global scope.
+// The period is in nanoseconds and represents the interval between adding new rows to the table.
+// The startTime is in nanoseconds since epoch and defaults to the current time when it is nil.
+func (ts *tableStub) TimeTable(ctx context.Context, period int64, startTime *int64) (*TableHandle, error) {
+	ctx, err := ts.client.withToken(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	result := ts.client.newTicket()
+
+	var realStartTime int64
+	if startTime == nil {
+		// TODO: Is this affected by timezones? Does it need to be the monotonic reading?
+		realStartTime = time.Now().UnixNano()
+	} else {
+		realStartTime = *startTime
+	}
+
+	req := tablepb2.TimeTableRequest{ResultId: &result, PeriodNanos: period, StartTimeNanos: realStartTime}
+	resp, err := ts.stub.TimeTable(ctx, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return parseCreationResponse(ts.client, resp)
+}
+
+// parseCreationResponse turns a gRPC table creation response, which is returned by most table gRPC methods, into a usable TableHandle.
+func parseCreationResponse(client *Client, resp *tablepb2.ExportedTableCreationResponse) (*TableHandle, error) {
+	if !resp.Success {
+		return nil, errors.New("server error: `" + resp.GetErrorInfo() + "`")
+	}
+
+	respTicket := resp.ResultId.GetTicket()
+	if respTicket == nil {
+		return nil, errors.New("server response did not have ticket")
+	}
+
+	alloc := memory.NewGoAllocator()
```

Comment:
As it turns out, arrow allocators are just wrappers around `make` and `copy`—the struct doesn't even contain any members. Nonetheless, instead of calling `memory.NewGoAllocator()` every time I need one, I now use the global variable `memory.DefaultAllocator`, since that seemed more clear (and less likely to have sudden performance costs in a future update of the Arrow API).